docs/Compile.rst

```diff
@@ -121,7 +156,9 @@ DF, which causes DF to use your system libstdc++ instead::
 Note that distributing binaries compiled with newer GCC versions requires end-
 users to delete libstdc++ themselves and have a libstdc++ on their system from
 the same GCC version or newer. For this reason, distributing anything compiled
-with GCC versions newer than 4.8 is discouraged.
```

Comment:
Sorry, I was primarily talking about distribution there (on Linux only). Pretty much any linux distro at this point should default to GCC 4.8 or newer, so compiling with GCC 4.5-4.8 should result in binaries that work on a majority of distros. We ran into issues in the past where Arch moved to GCC 4.9 but Debian stayed at 4.8, for example, and in that case GCC 4.9 wasn't even available in apt, so a couple DFHack builds broke completely on a lot of distros.
In short:
* GCC 4.5 produces builds that should work for everyone, but probably needs to be compiled from source or obtained from a third-party package repo
* GCC 4.6-4.8 (and maybe 4.9 by now?) produce builds that work for nearly everyone (with a new enough libstdc++), but users have to get rid of DF's libstdc++
I haven't investigated the availability of GCC 5 yet, but I'd say sticking with 4.5-4.8 (or 4.9) is best for releases for now.